yaptide/batch/string_templates.py

```diff
@@ -34,8 +34,7 @@
 ARRAY_SCRIPT=$ROOT_DIR/array_script.sh
 COLLECT_SCRIPT=$ROOT_DIR/collect_script.sh
 
-SHIELDHIT_CMD="sbatch --array=1-{n_tasks} --time=00:34:59\\
-    -A plgccbmc11-cpu --partition=plgrid --parsable $ARRAY_SCRIPT > $OUT"
+SHIELDHIT_CMD="sbatch --array=1-{n_tasks} {cmd_options} --parsable $ARRAY_SCRIPT > $OUT"
```

Comment:
Check here if this command executed with status code 0. Otherwise send back to user stdout and stderr. This will explain user if some unsupported options were parsed.

Reply:
Also add a comment or issue to check validity of command line arguments. We should have some basic protection from injection of code like `rm -rf`. Maybe chat gpt suggest something....

Reply:
Something like

```
import re

# Define a whitelist of allowed characters for cmd_options
allowed_chars = r'[\w\-.,=/:]+'

def sanitize_input(input_str):
    """Sanitize input string by removing any characters that are not in the allowed character set."""
    return re.sub(f'[^\s{allowed_chars}]', '', input_str)

# Example usage:
cmd_options = sanitize_input(user_input_cmd_options)
SHIELDHIT_CMD = f"sbatch --array=1-{n_tasks} {cmd_options} --parsable $ARRAY_SCRIPT > $OUT"
```

Reply:
Something like

```
import re

# Define a whitelist of allowed characters for cmd_options
allowed_chars = r'[\w\-.,=/:]+'

def sanitize_input(input_str):
    """Sanitize input string by removing any characters that are not in the allowed character set."""
    return re.sub(f'[^\s{allowed_chars}]', '', input_str)

# Example usage:
cmd_options = sanitize_input(user_input_cmd_options)
SHIELDHIT_CMD = f"sbatch --array=1-{n_tasks} {cmd_options} --parsable $ARRAY_SCRIPT > $OUT"
```

Reply:
> Also add a comment or issue to check validity of command line arguments. We should have some basic protection from injection of code like `rm -rf`. Maybe chat gpt suggest something....

Maybe we should prepare a list of supported options and skip unsupported ones?